src/main.rs

```diff
@@ -525,12 +618,19 @@ fn get_setup(args: &[String]) -> Setup {
 
         let mut volume_range = matches
             .opt_str(VOLUME_RANGE)
-            .map(|range| range.parse::<f64>().unwrap())
+            .map(|range| {
+                range.parse::<f64>().unwrap_or_else(|_| {
+                    error!("Invalid --{}: {}", VOLUME_RANGE, range);
+                    println!("Default: softvol - 60, alsa - what the control supports");
```

Comment:
For bonus points,  you could introduce a `MIXERS` constant in `mixer/mod.rs` and iterate over the available mixers, like the `audio_backend/mod.rs` does for the backends.

Reply:
Same question as above.

Reply:
Why iterate though? There's only 2 options? Seems silly to iterate when the choice is binary?

Reply:
Also another question is it reasonable to have a max range here? 96 would be a natural max as it's the dynamic range of 16 bit audio and probably more than enough but 100 also works because it's a nice round number.

Reply:
> Why iterate though? There's only 2 options? Seems silly to iterate when the choice is binary?

Like I said, it'd be for bonus points. I don't see any further mixers being added, but I like to do these things programatically instead of hard-coded. It's not a requirement for me, only something nice to have and I thought maybe easily copied from existing logic.



> Also another question is it reasonable to have a max range here? 96 would be a natural max as it's the dynamic range of 16 bit audio and probably more than enough but 100 also works because it's a nice round number.

100 dB is a good number. Note that this is more related to dB SPL than to DNR: https://www.dr-lex.be/info-stuff/volumecontrols.html